### PR TITLE
Rust helloworld example application

### PR DESCRIPTION
**Pull request by huyhoang8398**
add Rust helloworld programs

### CODE REVIEW

**Review by razvand — APPROVED** [review]
LGTM. Thanks, @huyhoang8398

File: lang/rust/Makefile [hunk 1]
```diff
@@ -0,0 +1,4 @@
+all: helloworld
+
+helloworld: helloworld.rs:
```
Comment (mkroening) [inline lang/rust/Makefile:3]: Without removing the trailing colon I get

```
$ make
Makefile:3: *** target pattern contains no '%'.  Stop.
```

```suggestion
helloworld: helloworld.rs
```

**Review by mkroening — COMMENTED** [review]
This App does not work on App ELF loader yet.

Apart from the Makefile not working, the resulting binary fails with

```
fatal runtime error: assertion failed: signal(libc::SIGPIPE, libc::SIG_IGN) != libc::SIG_ERR
```

It works on Linux, though.

The failing assert is here: [`std/src/sys/unix/mod.rs#L155`](https://github.com/rust-lang/rust/blob/c3605f8c8020dbbe8f0d1961c7b33c4c4b78ad0d/library/std/src/sys/unix/mod.rs#L155)

**Comment by razvand** [comment]
@huyhoang8398 , @mkroening , I'm not able to get it to build. I get the error:
```
error[E0463]: can't find crate for `std`
  |
  = note: the `x86_64-unknown-linux-musl` target may not be installed
  = help: consider downloading the target with `rustup target add x86_64-unknown-linux-musl`
```
I've little experience with Rust and I aimed to solve this by using `rustup` but I wasn't able to.

Could you please add a `README.md` file describing what needs to be installed / set up for this to work? Also, please add the `helloworld` executable file in the commit, a `clean` rule in the `Makefile` and, if required, a `.gitignore` file. See the support for other helloworld programs (such as [C](https://github.com/unikraft/static-pie-apps/tree/master/lang/c)).

**Comment by razvand** [comment]
This is superseded by #8 from @mkroening

**Pull request closed without merging**